cpp/src/arrow/compute/kernels/scalar_nested_test.cc

```diff
@@ -168,6 +173,20 @@ TEST(TestScalarNested, ListSliceFixedOutput) {
       expected = ArrayFromJSON(fixed_size_list(value_type, 2),
                                "[[3, null], [null, null], [null, null], null]");
       CheckScalarUnary("list_slice", input, expected, &args);
+
+      args.start = 1;
+      args.stop = std::nullopt;
+      expected = ArrayFromJSON(fixed_size_list(value_type, 2),
+                               "[[2, 3], [5, null], [null, null], null]");
+      if (input->type()->id() == Type::FIXED_SIZE_LIST) {
+        CheckScalarUnary("list_slice", input, expected, &args);
+      } else {
+        EXPECT_RAISES_WITH_MESSAGE_THAT(
+            NotImplemented,
+            ::testing::HasSubstr("Unable to produce FixedSizeListArray from "
+                                 "non-FixedSizeListArray without `stop` being set."),
+            CallFunction("list_slice", {input}, &args));
+      }
```

Comment:
Is this a case that is possible?

```suggestion
      }
      
      args.start = 3;
      args.stop = std::nullopt;
      expected = ArrayFromJSON(fixed_size_list(value_type, 0),
                               "[[], [], [], null]");
      if (input->type()->id() == Type::FIXED_SIZE_LIST) {
        CheckScalarUnary("list_slice", input, expected, &args);
      }
```

Reply:
That can indeed be possible, and a nice extra test case. Thank you!